src/confluent_kafka/src/Consumer.c

```diff
@@ -1341,10 +1458,16 @@ static void Consumer_rebalance_cb (rd_kafka_t *rk, rd_kafka_resp_err_t err,
 			return;
 		}
 
-		result = PyObject_CallObject(
-			err == RD_KAFKA_RESP_ERR__ASSIGN_PARTITIONS ?
-			self->u.Consumer.on_assign :
-			self->u.Consumer.on_revoke, args);
+                if (err == RD_KAFKA_RESP_ERR__ASSIGN_PARTITIONS)
+                        cb = self->u.Consumer.on_assign;
+                else {
+                        if (rd_kafka_assignment_lost(rk) &&
```

Comment:
nit: This nested if doesn't provide anything apart from intend, do else if + else

Reply:
nit: the structure I have better expresses the logic here imo, so preferred as being easier to read even though this could be written out more tersely as you suggest.